pkg/security/ebpf/c/include/structs/rate_limiter.h

```diff
@@ -2,10 +2,33 @@
 #define _STRUCTS_RATE_LIMITER_H_
 
 struct rate_limiter_ctx {
-    u64 current_period;
-    u32 counter;
-    u32 padding;
+    /*
+        data is representing both the `current_period` start
+        in the first 7 bytes (basically current_period & ~0xff)
+        and the counter in the last byte
+    */
+    u64 data;
```

Comment:
I love the optimization! I have a single remark: now the rate will be limited to 254, which is way less than the actual default of 500 (for AD). An u32 rate is given to `activity_dump_rate_limiter_allow` which will be down casted to u8 :/